src/librustc/ty/print/pretty.rs

```diff
@@ -213,6 +213,23 @@ pub trait PrettyPrinter<'tcx>:
         Ok(self)
     }
 
+    /// Prints `{...}` around what `f` and optionally `t` print
+    fn type_ascribed_value(
+        mut self,
+        f: impl FnOnce(Self) -> Result<Self, Self::Error>,
+        t: impl FnOnce(Self) -> Result<Self, Self::Error>,
+        print_ty: bool,
+    ) -> Result<Self::Const, Self::Error> {
+        self.write_str("{")?;
+        self = f(self)?;
+        if print_ty {
+            self.write_str(": ")?;
+            self = t(self)?;
+        }
```

Comment:
But if `print_ty` is false, then it's just braces around whatever `f` prints and this method isn't needed.